pkg/controller/installation/validation.go

```diff
@@ -187,102 +181,31 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 						}
 					default:
 						return fmt.Errorf("%s is invalid for ipPool.encapsulation, should be one of %s",
-							v4pool.Encapsulation, strings.Join(operatorv1.EncapsulationTypesString, ","))
+							pool.Encapsulation, strings.Join(operatorv1.EncapsulationTypesString, ","))
 					}
 				case operatorv1.IPAMPluginHostLocal:
-					// Verify the specified encapsulation type is valid.
-					switch v4pool.Encapsulation {
+					// The host-local IPAM plugin doens't support VXLAN.
+					switch pool.Encapsulation {
 					case operatorv1.EncapsulationVXLAN, operatorv1.EncapsulationVXLANCrossSubnet:
 						return fmt.Errorf("%s is invalid for ipPool.encapsulation with %s CNI and %s IPAM",
-							v4pool.Encapsulation,
+							pool.Encapsulation,
 							instance.Spec.CNI.Type,
 							instance.Spec.CNI.IPAM.Type)
 					}
 				}
 			} else {
-				// Verify the specified encapsulation type is valid.
-				switch v4pool.Encapsulation {
+				// If not using Calico CNI, then the encapsulation must be None and BGP must be disabled.
+				switch pool.Encapsulation {
 				case operatorv1.EncapsulationNone:
 				default:
 					return fmt.Errorf("%s is invalid for ipPool.encapsulation when using non-Calico CNI, should be None",
-						v4pool.Encapsulation)
+						pool.Encapsulation)
 				}
 				if instance.Spec.CalicoNetwork.BGP != nil && *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPEnabled {
 					return fmt.Errorf("BGP is not supported when using non-Calico CNI")
 				}
-				if v4pool.NodeSelector != "all()" {
-					return fmt.Errorf("ipPool.nodeSelector (%s) should be 'all()'", v4pool.NodeSelector)
-				}
 			}
 
-			if v4pool.NodeSelector == "" {
-				return fmt.Errorf("ipPool.nodeSelector should not be empty")
-			}
-
-			// Verify NAT outgoing values.
-			switch v4pool.NATOutgoing {
-			case operatorv1.NATOutgoingEnabled, operatorv1.NATOutgoingDisabled:
-			default:
-				return fmt.Errorf("%s is invalid for ipPool.natOutgoing, should be one of %s",
-					v4pool.NATOutgoing, strings.Join(operatorv1.NATOutgoingTypesString, ","))
-			}
-
-			if v4pool.BlockSize != nil {
-				if *v4pool.BlockSize > 32 || *v4pool.BlockSize < 20 {
-					return fmt.Errorf("ipPool.blockSize must be greater than 19 and less than or equal to 32")
-				}
-
-				// Verify that the CIDR contains the blocksize.
-				ones, _ := cidr.Mask.Size()
-				if int32(ones) > *v4pool.BlockSize {
-					return fmt.Errorf("IP pool size is too small. It must be equal to or greater than the block size.")
-				}
-			}
-		}
-
-		if v6pool != nil {
```

Comment:
Refactored this into a loop above, which saved a bunch of lines of code since many of these checks were duplicated between IPv4 and IPv6 previously.